### ISSUE TITLE
Cancel/reset queries through API?

### ISSUE THREAD

**Issue by jc-harrison** [Flowminder/FlowKit#2035]
The query state machine includes a 'cancelled' state, but there's currently no way for an API user to cancel a running query (e.g. if they made a mistake or realise it will take too long to calculate).

Similarly, if a query is in 'cancelled' or 'errored' state, it must be reset before it can be run again, but there's no way to reset a query through the API.

It would be useful to be able to do both of these things, although we'd need to think about how to handle the permissions.

**Comment by greenape** [comment]
Yeah. The permissions are a bit tricky - what you don't want is users to be able to cancel each other's queries (unless they're some sort of superuser).
